### PR TITLE
[JN-487] admin left bar

### PR DESCRIPTION
**Pull request by devonbush**
This takes a first pass at implementing Erin's designs at https://dspuxteam.invisionapp.com/console/share/UQE7Z456RK2/986004503.  there are plenty of features still missing, but this seemed like the best self-contained section of work that makes the admin tool materially better.  There's a lot of trivial route refactoring and markup rearrangement, the major changes to pay attention to are:

1.  The overall navigation approach is shifting from portal-centric to study-centric.  So rather than the opening screen being select a portal, it's now selecting a study.
2.  There is now an environment selector and a study selector, which allows changing the study/environment shown without having to navigate up/down in the hierarchy first.  This means that we have to be much more careful about our useEffect dependencies  on page loads--if we're loading data that's study-specific, we need to make sure the useEffect triggering the load retriggers if the study is changed.  This is probably the most likely source of serious bugs in this PR, after misplaced links.
3. the new sidebar makes it possible to massively simplify a bunch of other pages, especially the participant list 
image

TO TEST:
1.  populate 2+ portals in your local environment
2. go to localhost:3000
3. generally click around, exercise the left navbar 
4. when it appears, exercise the study and environment selectors, making sure data refreshes on selections

### CODE REVIEW

**Comment by breilly2** [comment]
Bug: "Forms & Surveys" is appearing active when any item in the "Design & Build" section is active.
Screenshot 2023-08-09 at 11 49 02 AM
In this screenshot, I'm viewing the "Website" section.
It also steals active state when switching environments in other sections, such as viewing the sandbox participant list.

**Comment by breilly2** [comment]
When viewing a participant, the shortcode part of the top breadcrumb nav is broken.
Screenshot 2023-08-09 at 11 55 05 AM
Text shouldn't be white, and the link goes to https://localhost:3000/ourhealth/studies/ourheart/env/sandbox/enrollees/OHSALK (`enrollees` instead of `participants`).

**Comment by breilly2** [comment]
Clicking the website link in the "Publish content" section goes to https://localhost:3000/ourhealth/env/sandbox/siteContent instead of https://localhost:3000/ourhealth/studies/ourheart/env/sandbox/siteContent, resulting in an empty left nav.
Screenshot 2023-08-09 at 12 00 55 PM
Screenshot 2023-08-09 at 12 01 08 PM

Side note/nit: While typing out this comment, I noticed that there's a mix of capitalization conventions in the left nav items.

**Comment by breilly2** [comment]
Clicking any of the links in the "Emails & Notifications" section results in "Unknown study environment page".

**Comment by breilly2** [comment]
On the participant list, if the page gets very wide because of many columns being visible, the help and account icons in the top nav end up off-screen.
Screenshot 2023-08-09 at 12 13 14 PM
In the screenshot, scrolled horizontally to see them. Where the gray background turns to white represents what I would see without scrolling.

Related note: When I sent an ad-hoc email, the toast notification that the email was sent successfully rendered in the top-right corner of the window as expected, and stayed fixed to that corner when scrolling horizontally.

File: ui-admin/src/navbar/StudySidebar.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+          <NavLink to={studyParticipantsPath(portalShortcode, study.shortcode, 'live')}
+            className="text-white p-1 rounded" style={navStyleFunc}>Participant List</NavLink>
+        </li>
+        <li className="mb-3">
+          <NavLink to={studyKitsPath(portalShortcode, study.shortcode, 'live')}
+            className="text-white p-1 rounded" style={navStyleFunc}>Biologistics</NavLink>
+        </li>
+        <li>
+          <NavLink to={studyEnvMailingListPath(portalShortcode, study.shortcode, 'live')}
+            className="text-white p-1 rounded" style={navStyleFunc}>Mailing list</NavLink>
```
Comment (breilly2) [inline ui-admin/src/navbar/StudySidebar.tsx]:
```suggestion
            className="text-white p-1 rounded" style={navStyleFunc}>Mailing List</NavLink>
```

File: ui-admin/src/navbar/StudySidebar.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        </li>
+      </ul>}/>
+      <CollapsableMenu header={'Design & Build'} content={<ul className="list-unstyled">
+        <li className="mb-3">
+          <NavLink to={studyEnvSiteContentPath(portalShortcode, study.shortcode, 'sandbox')}
+            className="text-white p-1 rounded" style={navStyleFunc}>Website</NavLink>
+        </li>
+        <li className="mb-3">
+          <NavLink to={studyEnvFormsPath(portalShortcode, study.shortcode, 'sandbox')}
+            className="text-white p-1 rounded" style={navStyleFunc}>Forms & Surveys</NavLink>
```
Comment (breilly2) [inline ui-admin/src/navbar/StudySidebar.tsx]: Nit, but I've noticed you've been deliberate about using `&amp;` in other places.
```suggestion
            className="text-white p-1 rounded" style={navStyleFunc}>Forms &amp; Surveys</NavLink>
```

File: ui-admin/src/navbar/StudySidebar.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        </li>
+        <li>
+          <NavLink to={studyEnvNotificationsPath(portalShortcode, study.shortcode, 'sandbox')}
+            className="text-white p-1 rounded" style={navStyleFunc}>Emails & Notifications</NavLink>
```
Comment (breilly2) [inline ui-admin/src/navbar/StudySidebar.tsx]:
```suggestion
            className="text-white p-1 rounded" style={navStyleFunc}>Emails &amp; Notifications</NavLink>
```

**Review by breilly2 — APPROVED** [review]
Looks great! I spent more time testing and only skimmed over the code, but at a high level, it looks reasonable to me.

Definitely still some details to iron out. The previous design reflected the underlying 2-dimensional nature of the data. The new design is more opinionated about highlighting specific areas, such as live participants and sandbox site content. I think that somewhat obscures the 2-dimensional nature of the model, which could make it harder for a newcomer to understand that model; they may get confused or feel lost if they find themselves in less-used corners, such as sandbox participants.

**Review by JeffKorte — APPROVED** [review]
After walking through this with Devon on Aug 9, this looks good to move to Dev.

**Pull request merged**